Library/Homebrew/test/rubocops/components_order_spec.rb

```diff
@@ -523,13 +545,64 @@ class Foo < Formula
       RUBY
     end
 
+    it "reports no offenses if the on_macos block has if-else branches and they are properly formatted" do
+      expect_no_offenses(<<~RUBY)
+        class Foo < Formula
+          url "https://brew.sh/foo-1.0.tgz"
+
+          resource do
+            on_macos do
+              if foo == :bar
+                url "https://brew.sh/resource2.tar.gz"
+                sha256 "586372eb92059873e29eba4f9dec8381541b4d3834660707faf8ba59146dfc35"
+              else
+                url "https://brew.sh/resource1.tar.gz"
+                sha256 "686372eb92059873e29eba4f9dec8381541b4d3834660707faf8ba59146dfc35"
+              end
+            end
+
+            on_linux do
+              url "https://brew.sh/resource2.tar.gz"
+              sha256 "586372eb92059873e29eba4f9dec8381541b4d3834660707faf8ba59146dfc35"
+            end
+          end
+        end
+      RUBY
+    end
+
+    it "the content of the on_macos block is wrong and not a method" do
+      expect_offense(<<~RUBY)
+        class Foo < Formula
+          url "https://brew.sh/foo-1.0.tgz"
+
+          resource do
+          ^^^^^^^^^^^ `on_macos` blocks within resource blocks must contain only a url and sha256 or a url, version, and sha256 (in those orders).
+            on_macos do
+              if foo == :bar
+                url "https://brew.sh/resource2.tar.gz"
+                sha256 "586372eb92059873e29eba4f9dec8381541b4d3834660707faf8ba59146dfc35"
+              else
+                sha256 "686372eb92059873e29eba4f9dec8381541b4d3834660707faf8ba59146dfc35"
+                url "https://brew.sh/resource1.tar.gz"
+              end
+            end
+
+            on_linux do
+              url "https://brew.sh/resource2.tar.gz"
+              sha256 "586372eb92059873e29eba4f9dec8381541b4d3834660707faf8ba59146dfc35"
+            end
+          end
+        end
+      RUBY
+    end
+
     it "the content of the on_linux block is wrong in a resource block" do
```

Comment:
One thing I noticed was that a lot of the tests in the `tests/rubocops` directory don't have very good descriptions for `it` and `context` blocks. (I don't blame you @Rylan12 in this particular PR since the surrounding descriptions here aren't very... good)

For example, if the test above this comment fails, here's what `brew tests` spits out:

```
Failures:

  1) RuboCop::Cop::FormulaAudit::ComponentsOrder resource the content of the on_linux block is wrong in a resource block
```

In contrast, here's what is outputted when [this test in `test/utils/curl_spec`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/utils/curl_spec.rb#L11) fails:

```
Failures:

  1) Utils::Curl curl_args returns --disable as the first argument when HOMEBREW_CURLRC is not set
```

This is an issue that probably should be addressed in one or more future PRs (by willing volunteers, of course)

Reply:
I went ahead and fixed up just this section. Later, I can start going through and fixing up other sections as well.